src/dists.ad.js

```diff
@@ -616,12 +616,42 @@ var Binomial = makeDistributionType({
     'use ad';
     var p = this.params.p;
     var n = this.params.n;
-    // exact formula
-    return (lnfact(n) - lnfact(n - val) - lnfact(val) +
-            val * Math.log(p) + (n - val) * Math.log(1 - p));
+
+    if (!_.contains(this.support(), val)) {
```

Comment:
I'm not sure if it matters much, but we could make this a bit more performant for large `n` by avoiding building the entire list of support elements and instead just checking whether `val >= 0 && val <= n  && val % 1 == 0`.